[Beats CM] Add beat name to update endpoint

justinkambic opened the pull request:
## Changes
- Add `name` field to the `update` endpoint for the server
- Add `name` field to beat ES index template
- Write tests for the update domain function

mattapperson commented:
could you also add/update a test to the functional API test?

mattapperson commented:
Could you also update the beats tables to display the name va the id as it does currently

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/2925/)

justinkambic commented:
Showing name instead of ID
<img width="517" alt="image" src="https://user-images.githubusercontent.com/18429259/45760028-2cef5a00-bbf7-11e8-8159-5f407f902156.png">

I added `name` to the ES archives, and added a check for it in the `update` functional test. I also fixed one of the broken tests, there are several others that are broken, but they're broken on the main branch as well, so I didn't want to go down the rabbit hole of fixing them all in this PR.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/2948/)

exekias commented:
I opened https://github.com/elastic/beats/pull/8380 to send the `name` field when enrolling

justinkambic commented:
I also want to note before merging this that we have updated the `scripts/enroll.js` script to generate a name when enrolling beats for development/smoke testing purposes.
